Add externl references to NAP (#1100)
nginxinc/kubernetes-ingress: docs-web/app-protect/troubleshooting.md

     - The Ingress Controller NGINX verification timeouts while starting for the first time or while reloading after a change.
     - Check the logs for ``Unable to fetch version: X`` message.
     - Too many Ingress Resources with App Protect enabled. Check the `NGINX fails to start/reload section <#nginx-fails-to-start-or-reload>`_ of the Known Issues. 

>    * - Reload time.

Reviewer: reload time is already covered by the previous row. Better to extend the Troubleshooting method column with an additional sentence: Check the HTTP response time to resources used in App Protect policy external references. Note that on its own, external references is not the only reason for increased reload. That's why the suggestion.

     - The Ingress Controller NGINX Configuration timeout takes longer than usual.

>      - Check the HTTP resonse time to resources used in App Protect policy external references.

Reviewer 2: ```suggestion
     - Check the HTTP response time to resources used in App Protect policy external references.
```

- You need to apply a large amount of Ingress Resources at once.
- You are running the Ingress Controller for the first time in a cluster where the Ingress Resources with App Protect enabled are already present.

You can increase this timeout by setting the `nginx-reload-timeout` [cli-argument](/nginx-ingress-controller/configuration/global-configuration/command-line-arguments/#cmdoption-nginx-reload-timeout).

> If You are using external references in Your Nginx App Protect policies, verify if the servers hosting the referenced resources are available and that their response time is as short as possible (see next paragraph).

Reviewer 2: ```suggestion
If you are using external references in your Nginx App Protect policies, verify if the servers hosting the referenced resources are available and that their response time is as short as possible (see next paragraph).
```

> If you are using external references in your Nginx App Protect policies, verify if the servers hosting the referenced resources are available and that their response time is as short as possible (see next paragraph).

Reviewer: ```suggestion
If you are using external references in your Nginx App Protect policies, verify if the servers hosting the referenced resources are available and that their response time is as short as possible (see the Check the HTTP response time to resources used in App Protect policy external references section).
```

> If the references are not available during ingress controller startup the pod will fail to startup. In case the resources are not available during a reload, the relod will fail, and ingress controller will process traffic with the previous correct coonfiguration.

Reviewer 2: ```suggestion
If the references are not available during ingress controller startup the pod will fail to startup. In case the resources are not available during a reload, the reload will fail, and ingress controller will process traffic with the previous correct configuration.
```

> If the references are not available during ingress controller startup the pod will fail to startup. In case the resources are not available during a reload, the reload will fail, and ingress controller will process traffic with the previous correct configuration.

Reviewer: there is no need for a line break here. it is not preserved in the generated docs:
![image](https://user-images.githubusercontent.com/2199726/90543778-087dbe80-e13b-11ea-86ef-c849a1a16acd.png)

> If the references are not available during ingress controller startup the pod will fail to startup. In case the resources are not available during a reload, the reload will fail, and ingress controller will process traffic with the previous correct configuration.

Reviewer: ```suggestion
If the references are not available during the Ingress Controller startup, the pod will fail to start. In case the resources are not available during a reload, the reload will fail, and NGINX Plus will use the previous correct configuration.
```

> ### Check the HTTP resonse time to resources used in App Protect policy external references.

Reviewer 2: ```suggestion
### Check the HTTP response time to resources used in App Protect policy external references.
```

> ### Check the HTTP response time to resources used in App Protect policy external references.

Reviewer: This should be moved under the Troubleshooting Methods section, as it is a troubleshooting method.

I suggest two small enhancements:
* For brevity, renaming it to "Check the Availability of APPolicy External References" 
* Adding the disclaimer like this:
This method only applies if you're using [external references](https://docs.nginx.com/nginx-app-protect/configuration/#external-references) in NGINX App Protect policies.

> To check what servers host the external references: 

Reviewer: ```suggestion
To check what servers host the external references of a policy: 
```
